lte/gateway/c/oai/tasks/nas/api/mme/mme_api.c

```diff
@@ -621,3 +624,14 @@ int mme_api_unsubscribe(bstring apn) {
   _mme_api_pdn_id -= 1;
   OAILOG_FUNC_RETURN(LOG_NAS, rc);
 }
+
+uint32_t generateRandomTMSI(struct ue_mm_context_s* ue_context) {
+  /*
+   * Generate TMSI with a combination of memory address of ue_context and
+   * a random number to avoid possible collisond (due to UE holding to a
+   * specific TMSI
+   */
+  tmsi_t tmsi = rand() & 0xff;
+  tmsi        = tmsi + (tmsi_t)(uintptr_t) ue_context;
```

Comment:
Why do we need ue_context address all together?
Why not simply use
`return (tmsi_t) rand();`

Reply:
I didn't want to use rand by itself because the psedo-random generator repeat after some time. This is a big number, but if we have some UEs that are always connected and they never release the session, we could end up with duplication. 

The usage of memory address plus random number may reduce the chance of this happening on sticky sessions. 

But now that we see other contributors are reporting this, we may need to reconsider

Reply:
I dont anticipate the psedo-random generator to repeat in my life time :)
I dont see how adding the address improves the odds
I would rather have the cleaner code